### PR TITLE
[FEATURE] Add release script

### PR DESCRIPTION
**Pull request by kevinmcdonnell**

### CODE REVIEW

**Comment by shtrom** [comment]
Sorry @kevinmcdonnell , didn't get a chance to get to it yet. From a quick glance it looks good.

It might be beneficial to run `shellcheck` on it, as it flags a lot of sub-optimal patterns (I haven't seen any, but it will be more exhaustive than me). It's in Homebrew.

It would be good to add a shebang, and make the script executable, too, so the Makefile doesn't have to specificy it needs to run through `bash`.

I'll review in more details soon.

File: release.sh [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+get_branch () {
+	# Get the branch to release and abort if branch does not exist or is not checked out
+	echo "What branch do you want to release?"
+	read branch
+	git rev-parse --verify $branch
+	current_branch=`git rev-parse --abbrev-ref HEAD`
+
+	if ! [[ $current_branch = $branch ]]
```
Comment (shtrom) [inline release.sh]: If we can only support release the current branch (which is fine, I think), maybe we should just show the user what the branch is, and ask them whether that's what they want.

File: release.sh [hunk 2]
```diff
@@ -0,0 +1,91 @@
+set -e
+
+check_version () {
+	if ! [[ $1 =~ ^v[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$ ]]
+	then
+	    echo -e "\nThat version does not match semver"
+	    exit
```
Comment (shtrom) [inline release.sh]: Maybe we should just return 0 (success) or !=0 (e.g., 1; failure), so the caller can offer the user to enter the tag again.

File: release.sh [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+get_hash () {
+	current_hash=`git rev-parse HEAD`
+}
+
+get_new_version () {
+	# Get new version to release
+	echo -e "\nWhat version do you want to release?"
+	read new_version
+	check_version $new_version
```
Comment (shtrom) [inline release.sh]: If `check_version` returns 0 or 1, we can do something like
```bash
while ! check_version $new_version; do
    read -p "New version: " new_version
done
```

File: release.sh [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+get_prev_version () {
+	# Get previous version to generate release notes
+	echo -e "\nWhat is the previous version? (used to generate release notes)"
+	read prev_version
+	check_version $prev_version
```
Comment (shtrom) [inline release.sh]: Ditto, but we should also check that `$prev_version` is a tag.

File: release.sh [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+	fi
+}
+
+get_hash () {
+	current_hash=`git rev-parse HEAD`
+}
+
+get_new_version () {
+	# Get new version to release
+	echo -e "\nWhat version do you want to release?"
```
Comment (shtrom) [inline release.sh]: We could use `read -p` here.

File: release.sh [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+	then
+		echo "Aborting..."
+		exit
+	fi
+}
+
+update_version () {
+	# update and commit local version file used by tracking telemetry
+	echo -e "\nWriting version file..."
+	echo $new_version > version
```
Comment (shtrom) [inline release.sh]: For the PHP SDK, this should be `.version` at the root.

File: release.sh [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+	echo -e "\nCommitting version file..."
+	git add version
+	git commit -m "[BUILD] Update version with latest release"
+	git push origin $branch
+}
+
+push_tag () {
+	# push version to github
+	echo -e "\nTagging..."
+	git tag $new_version $branch -m "$release_notes"
```
Comment (shtrom) [inline release.sh]: I'd got for an annotated tag with our usual format for the first line. `git` supports multiple `-m` arguments.

`git tag "${new_version}" "${branch}" -a -m "[RELEASE] ${new_version}" -m "${release_notes}"`

File: release.sh [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+update_version () {
+	# update and commit local version file used by tracking telemetry
+	echo -e "\nWriting version file..."
+	echo $new_version > version
+	
+	echo -e "\nCommitting version file..."
+	git add version
+	git commit -m "[BUILD] Update version with latest release"
```
Comment (shtrom) [inline release.sh]: I think we could just use `"[RELEASE] ${new_version}"` as the commit message here.

**Comment by kevinmcdonnell** [comment]
@shtrom ready for another look and test

File: release.sh [hunk 9]
```diff
@@ -0,0 +1,91 @@
+set -e
```
[the remark this answers is not in the document]
Comment (shtrom) [reply release.sh]: Actually, you can add the `-e` to the shebang.

**Comment by kevinmcdonnell** [comment]
@shtrom would you like to add these changes? not sure it's efficient for me to do so..

**Comment by kevinmcdonnell** [comment]
shellcheck said to use double backslash

**Pull request merged**